### PR TITLE
chore(recordings): check target's jvmId for recording events

### PR DESCRIPTION
**Pull request by mwangggg**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits using a GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification)

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Depends on https://github.com/cryostatio/cryostat/pull/1689
Related to https://github.com/cryostatio/cryostat/issues/1452

### CODE REVIEW

File: src/app/Agent/AgentLiveProbes.tsx [hunk 1]
```diff
@@ -210,7 +210,7 @@ export const AgentLiveProbes: React.FC<AgentLiveProbesProps> = (_) => {
         context.target.target(),
         context.notificationChannel.messages(NotificationCategory.ProbeTemplateApplied),
       ]).subscribe(([currentTarget, e]) => {
-        if (currentTarget?.connectUrl != e.message.targetId) {
+        if (currentTarget?.connectUrl != e.message.targetId && currentTarget?.jvmId != e.message.jvmId) {
```
Comment (tthvo) [inline src/app/Agent/AgentLiveProbes.tsx:213]: Would we just need to check `jvmId`?

**Review by tthvo — COMMENTED** [review]
Topology view needs updates too.

https://github.com/cryostatio/cryostat-web/blob/86a8094c1947004fb1f625da7f517d071e2920d1/src/app/Topology/Entity/utils.tsx#L349

I would recommend doing a global search for these target-owned resources' notification category and ensure all are also or just using jvmId for checking.

File: src/app/Agent/AgentLiveProbes.tsx (hunk 1 above)
Comment (mwangggg) [reply src/app/Agent/AgentLiveProbes.tsx:213]: updated to just check jvmId for every event

**Comment by mwangggg** [comment]
had to re-add the connectUrl check in notification messages because the integration tests were failing without them

**Comment by mwangggg** [comment]
https://github.com/cryostatio/cryostat-web/actions/runs/6592311210

**Comment by mwangggg** [comment]
https://github.com/cryostatio/cryostat-web/actions/runs/6615277073

**Review by andrewazores — APPROVED** [review]

**Pull request merged**